### PR TITLE
fix(#360) - Additional field for permissions in the Home Service Schema

### LINKED ISSUE #360: Additional field for permissions in the Home Service Schema
```
permissions Array[]:
    roverGroup: String
    role: enum("ADMIN", "USER")
```

## Describe the solution you'd like

- The HomeType schema should have a field for storing rover-groups and their permissions for each SPA/Microservice.
- It should also provide mutations to add/edit the API permissions for each SPA/Microservice
- The permissions fields should be of type array, with the children being object of the type

## Describe alternatives you've considered

None

### PR DESCRIPTION
**Pull request by hybridx**
fix(#360) - Added permissions array to the schema, type.d.ts and type def.graphql

# Closes/Fixes/Resolves

### Resolves #360 

# Explain the feature/fix

Added permissions array to the schema 

## Does this PR introduce a breaking change

No

## Screenshots

Screenshot 2020-06-01 at 6 15 34 PM

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

File: packages/home-service/src/typedef.graphql [hunk 1]
```diff
@@ -32,6 +36,11 @@ type HomeType {
     createdOn: DateTime
     updatedBy: UserType
     updatedOn: DateTime
+    permissions: [PermissionsType]
```
Comment (deshmukhmayur) [inline packages/home-service/src/typedef.graphql:39]: Decrease indent

```suggestion
  createdOn: DateTime
  updatedBy: UserType
  updatedOn: DateTime
  permissions: [PermissionsType]
```

File: packages/home-service/src/types.d.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 declare module '*.graphql';
 declare module '*.json';
 
+type PermissionsType = {
+  roverGroup: string;
+  role: string;
```
Comment (deshmukhmayur) [inline packages/home-service/src/types.d.ts:6]: Use typescript string enums: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

File: packages/home-service/src/schema.ts [hunk 3]
```diff
@@ -13,6 +13,12 @@ export const HomeServiceSchema: Schema = new Schema({
     createdOn: Date,
     updatedBy: String,
     updatedOn: Date,
+    permissions: [
+        {
+            roverGroup: String,
+            role: String,
```
[the remark this answers is not in the document]
Comment (hybridx) [reply packages/home-service/src/schema.ts:19]: Added to schema
```
    permissions: [
        {
            roverGroup: String,
            role: {
                type: String,
                enum: ['ADMIN', 'USER'],
                required: true,
            }
        }
    ],
```

File: packages/home-service/src/typedef.graphql (hunk 1 above)
Comment (hybridx) [reply packages/home-service/src/typedef.graphql:39]: Added consistent indention

File: packages/home-service/src/types.d.ts (hunk 2 above)
Comment (hybridx) [reply packages/home-service/src/types.d.ts:6]: Added enum 
```
declare enum Role {
  admin = "ADMIN",
  user = "USER"
}
```

File: packages/home-service/src/types.d.ts [hunk 4]
```diff
@@ -1,30 +1,39 @@
 declare module '*.graphql';
 declare module '*.json';
 
+declare enum Role {
+  admin = "ADMIN",
+  user = "USER"
```
Comment (deshmukhmayur) [inline packages/home-service/src/types.d.ts:6]: Keep it all caps, for maintaining consistency between the typedef.graphql and typescript types.
```suggestion
  ADMIN = "ADMIN",
  USER = "USER"
```

**Pull request merged**